src/charm.py

```diff
@@ -667,14 +785,14 @@ def _get_configs_per_unit(self, relation: Relation) -> Optional[dict]:
                 )
                 continue
 
-            prefix = self._get_prefix(data)
+            prefix = self._get_prefix(data)  # type: ignore
```

Comment:
The `type: ignore` is becoming a burden.
- Would pinning pyright help?
- Is there a pyright setting to error out on unused ignores?

Reply:
- pinning pyright would help.
- yes, and it is the default.

Reply:
(I'm pinning pyright as part of this pr)